cmd/kubeadm/app/componentconfigs/defaults.go

```diff
@@ -88,7 +91,7 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 	// Enforce security-related kubelet options
 
 	// Require all clients to the kubelet API to have client certs signed by the cluster CA
-	externalkubeletcfg.Authentication.X509.ClientCAFile = kubeadmapiv1beta1.DefaultCACertPath
+	externalkubeletcfg.Authentication.X509.ClientCAFile = fmt.Sprintf("%s/%s", strings.TrimSuffix(internalcfg.CertificatesDir, "/"), constants.CACertName)
```

Comment:
Looks way better than before, however let's use `filepath.Join` for the actual path construction.

Reply:
Thank you so much! 
The code has been updated.